packages/jellyfish-api-core/__tests__/category/oracle/listLatestRawPrices.test.ts

```diff
@@ -0,0 +1,255 @@
+
+  it('should listLatestRawPrices', async () => {
+    const priceFeed1 = [
+      { token: 'APPLE', currency: 'EUR' },
+      { token: 'APPLE', currency: 'USD' }
+    ]
+    const priceFeed2 = [
+      { token: 'TESLA', currency: 'EUR' },
+      { token: 'TESLA', currency: 'USD' }
+    ]
+
+    const oracleid1 = await container.call('appointoracle', [await container.getNewAddress(), priceFeed1, 1])
+    const oracleid2 = await container.call('appointoracle', [await container.getNewAddress(), priceFeed2, 2])
+
+    await container.generate(1)
+
+    const timestamp1 = Math.floor(new Date().getTime() / 1000)
+    const timestamp2 = Math.floor(new Date().getTime() / 1000) + 300
+
+    const prices1 = [{ tokenAmount: '0.5@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid1, timestamp1, prices1])
+
+    const prices2 = [{ tokenAmount: '1@APPLE', currency: 'USD' }]
+    await container.call('setoracledata', [oracleid1, timestamp1, prices2])
+
+    const prices3 = [{ tokenAmount: '1.5@TESLA', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid2, timestamp2, prices3])
+
+    const prices4 = [{ tokenAmount: '2@TESLA', currency: 'USD' }]
+    await container.call('setoracledata', [oracleid2, timestamp2, prices4])
+
+    await container.generate(1)
+
+    // NOTE(jingyi2811): Pagination is not supported.
+    const data = await client.oracle.listLatestRawPrices()
+
+    const result1 = data.filter(element => element.oracleid === oracleid1)
+
+    expect(result1).toStrictEqual(
+      [
+        {
+          priceFeeds: priceFeed1[0],
+          oracleid: oracleid1,
+          weightage: new BigNumber(1),
+          timestamp: new BigNumber(timestamp1),
+          rawprice: new BigNumber(0.5),
+          state: OracleRawPriceState.LIVE
+        },
+        {
+          priceFeeds: priceFeed1[1],
+          oracleid: oracleid1,
+          weightage: new BigNumber(1),
+          timestamp: new BigNumber(timestamp1),
+          rawprice: new BigNumber(1),
+          state: OracleRawPriceState.LIVE
+        }
+      ]
+    )
+
+    const result2 = data.filter(element => element.oracleid === oracleid2)
+
+    expect(result2).toStrictEqual(
+      [
+        {
+          priceFeeds: priceFeed2[0],
+          oracleid: oracleid2,
+          weightage: new BigNumber(2),
+          timestamp: new BigNumber(timestamp2),
+          rawprice: new BigNumber(1.5),
+          state: OracleRawPriceState.LIVE
+        },
+        {
+          priceFeeds: priceFeed2[1],
+          oracleid: oracleid2,
+          weightage: new BigNumber(2),
+          timestamp: new BigNumber(timestamp2),
+          rawprice: new BigNumber(2),
+          state: OracleRawPriceState.LIVE
+        }
+      ]
+    )
+
+    await container.call('removeoracle', [oracleid1])
+    await container.call('removeoracle', [oracleid2])
+
+    await container.generate(1)
+  })
+
+  it('should listLatestRawPrices with various timestamps', async () => {
```

Comment:
I would prefer you write 5 tests with 1 describe block instead of putting it all together, easier to read.

Reply:
It was like this
- test with the current timestamp
- test with 3599 seconds after the current timestamp
- test with 3600 seconds after the current timestamp
- test with 3599 seconds before the current timestamp
- test with 3600 seconds before the current timestamp

If I run all these tests 5 times, probably all would pass. However, it doesn't guarantee it will pass if I run all the tests 6th time.

Alter discussing with Ahmed a few days ago, he advised me to change the magic number 3600 to 4200.

In the new code, I have reduced the no of tests to 2.
- test with 4200 seconds after the current timestamp
- test with 4200 seconds before the current timestamp